backend/authentication/adminPasswordConfirm.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+const AWS = require('aws-sdk');
+const db = new AWS.DynamoDB.DocumentClient();
+const uuid = require('uuid');
+require('dotenv').config();
+
+async function adminPasswordConfirm(event, context, callback) {
+    try {
+        // Get the recovery code from db for this user
+        const recoveryCodes = await db.scan({
+            TableName: process.env.RECOVERY_CODES_TABLE
+        }).promise();
+
+        let recoveryEntry;
+        for (let i = 0; i < recoveryCodes.Items.length; i++) {
+            if (recoveryCodes.Items[i].code == event.arguments.confirmationCode) {
+                recoveryEntry = recoveryCodes.Items[i];
+            }
+        }
+        // Check if recovery entry exists in db
+        if (!recoveryEntry) {
+            let error = new Error;
+            error.message = 'User not found!!!';
+            return callback(error, null);
+        }
+
+        // Check if confirmation code sent by user is valid
+        if (recoveryEntry.code != event.arguments.confirmationCode) {
+            let error = new Error;
+            error.message = 'Confirmation code is not valid';
+            return callback(error, null);
+        }
+
+        //get admin for authentication
+        const admin = await db.get({
+            TableName: process.env.ADMINS_TABLE,
+            Key: {
+                id: recoveryEntry.admin_id,
+            },
+        }).promise();
+        if(!admin) {
+            let error = new Error;
+            error.message = 'Failed to get admin from db';
+            return callback(error, null);
+        }
+        //return callback(null, `${admin.Item.email}`);
+        // update the password in cognito user pool
+        const cognito = new AWS.CognitoIdentityServiceProvider();
```

Comment:
Add static variables to the global scope of the function. Lambda functions do some caches with global variables. 
Also, be aware of using global variables. It can have the previous value because of caching.